adoc/chapters/programming_interface.adoc

```diff
@@ -21943,8 +21938,10 @@ shifting values a fixed number of work-items to the left or right.
 include::{header_dir}/algorithms/shift.h[lines=4..-1]
 ----
 
-  . _Constraints:_ Available only if [code]#std::is_same_v<std::decay_t<Group>,
-    sub_group># is true and [code]#T# is a trivially copyable type.
+  . _Constraints:_ Available only if all of the following conditions are met:
+    - [code]#sycl::is_same_v<std::decay_t<Group>, sub_group># is true; and
+    - [code]#T# is a trivially copyable type.
```

Comment:
Note that the spec does not guarantee that `vec` or `marray` are trivially copyable.  I believe there is an internal issue which suggests that we should change this.  At one point, DPC++ did **not** implement these as trivially copyable, though I think we may have fixed that by now.

Reply:
You're right, internal issue 564 talks about this.  I'll open a pull request to make these types trivially copyable, because I think that's the right fix.  I don't want to change the constraint here, because I think it's correct.